firewood/src/merkle.rs

```diff
@@ -1042,6 +1042,42 @@ impl<S: ShaleStore<Node> + Send + Sync, T> Merkle<S, T> {
         Ok(ptr.map(|ptr| RefMut::new(ptr, parents, self)))
     }
 
+    /// verify_proof checks merkle proofs. The given proof must contain the value for
+    /// key in a trie with the given root hash. VerifyProof returns an error if the
+    /// proof contains invalid trie nodes or the wrong value.
+    ///
+    /// The generic N represents the storage for the node data
+    pub fn verify_proof<K: AsRef<[u8]>, N: AsRef<[u8]> + Send>(
```

Comment:
Needed to move this method here because verifying a proof requires us to decode nodes, which we need a `merkle` to do now.